src/store/modules/share.store.js

```diff
@@ -22,9 +22,14 @@ export default {
     },
     getters: {},
     actions: {
-        async generateShortLink({ commit }) {
+        async generateShortLink({ commit }, withBalloonMarker = false) {
             try {
-                const shortLink = await createShortLink(window.location.href)
+                const shortLink = await createShortLink(
+                    // we do not want the geolocation of the user clicking the link to kick in, so we force the flag out of the URL
+                    window.location.href.replace('&geolocation=true', '') +
+                        // if the geolocation was being tracked by the user generating the link, we place a balloon (dropped pin) marker at his position (center of the screen, so no need to change any x/y position)
+                        (withBalloonMarker ? '&crosshair=marker' : '')
+                )
```

Comment:
Because you have to check and remove the `geolocation=true` here, would it not make more sense to do add the marker based on this ?
```javascript
async generateShortLink({ commit }) {
    let href = window.location.href.replace("&geolocation=true", "&crosshair=marker")
    try {
        const shortLink = await createShortLink(window.location.href)
        ...
    }
    ...
}
```
This way you don't have to pass any parameter and in the code it is clear that the url contains a `?`. Because in you code if for any reason the `withBalloonMarker` is set to `true` but the `href` doesn't contain any query param (I know this is very unlikely), the url will be brocken.

Reply:
the issue is, you could have activated your GPS, but not be tracking your position at the moment (you've panned somewhere else on the map, and want to share this portion of the map, not your current position).
That's why this balloon marker is placed only when `state.geolocation.active && state.geolocation.tracking`.

But we do not want the geolocation URL param to be passed along in any case

Reply:
and yes, placing a `&...=...` is "risky", but at this point in time of the app's life cycle, there's no way lat/lon/zoom or other mandatory params haven't been placed, so it's not that big of a risk

Reply:
Ok 
```